### PR TITLE
Filter kgraph percentile

### PR DESCRIPTION
**Pull request by finnagin**

### CODE REVIEW

**Review by dkoslicki — APPROVED** [review]

File: operations/filter_kgraph_percentile.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      minimum: 0
+      maximum: 100
+      default: 95
+    remove_above:
+      description: Indictes whether to remove above or below the given threshold.
+      type: bool
+      example: true
+      default: false
+    edge_keys:
+      description: This indicates if you only want to filter on specific edge_keys. If not provided or empty all edges will be filtered on and edge_key will not be considered when filtering.
```
Comment (patrickkwang) [inline operations/filter_kgraph_percentile.yml:40]:
```suggestion
      description: This indicates if you only want to filter on specific edge_keys. If not provided or empty, all edges will be filtered on.
```

File: operations/filter_kgraph_percentile.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      type: array
+      example: [e01]
+      default: null
+    qnode_keys:
+      description: This indicates if you only want nodes corresponding to a specific list of qnode_keys to be removed. If not provided oe empty no nodes will be removed when filtering.
```
Comment (patrickkwang) [inline operations/filter_kgraph_percentile.yml:45]:
```suggestion
      description: This indicates if you only want nodes corresponding to a specific list of qnode_keys to be removed. If not provided or empty, no nodes will be removed when filtering.
```

File: operations/filter_kgraph_percentile.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      example: normalized_google_distance
+    threshold:
+      description: The percentile to threshold on.
+      type: float
+      example: 96.8
+      minimum: 0
+      maximum: 100
+      default: 95
+    remove_above:
+      description: Indictes whether to remove above or below the given threshold.
```
Comment (patrickkwang) [inline operations/filter_kgraph_percentile.yml:35]:
```suggestion
      description: Indicates whether to remove above or below the given threshold.
```

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: docs/filter_kgraph_percentile.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    type: array
+  qnode_keys:
+    default: null
+    description: This indicates if you only want nodes corresponding to a specific
+      list of qnode_keys to be removed. If not provided oe empty no nodes will be
+      removed when filtering.
+    example:
+    - n01
+    type: array
+  remove_above:
```
Comment (dkoslicki) [inline docs/filter_kgraph_percentile.md:48]: `keep_above_or_below` with the options of `above` or `below` @finnagin

**Review by dkoslicki — CHANGES_REQUESTED** [review]

File: docs/filter_kgraph_percentile.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+### parameters
+
+```yaml
+properties:
+  edge_attribute:
+    description: The name of the edge attribute to filter on.
+    example: normalized_google_distance
+    type: string
+  edge_keys:
```
Comment (dkoslicki) [inline docs/filter_kgraph_percentile.md:32]: `qedge_keys`

**Review by dkoslicki — CHANGES_REQUESTED** [review]

File: docs/filter_kgraph_percentile.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      If not provided or empty all edges will be filtered on and edge_key will not
+      be considered when filtering.
+    example:
+    - e01
+    type: array
+  qnode_keys:
+    default: null
+    description: This indicates if you only want nodes corresponding to a specific
+      list of qnode_keys to be removed. If not provided oe empty no nodes will be
+      removed when filtering.
```
Comment (dkoslicki) [inline docs/filter_kgraph_percentile.md:44]: Add the following: Allows you to choose what happens to source and target nodes attached to edges that get filtered out. Only applicable to qnodes that are either sources or targets of edges with the provided `edge_attribute`.

**Review by dkoslicki — CHANGES_REQUESTED** [review]
Also applicable to all filters: remove all results that have bindings to edges/nodes filtered

**Review by dkoslicki — APPROVED** [review]

**Review by patrickkwang — CHANGES_REQUESTED** [review]

**Review by cbizon — APPROVED** [review]

**Review by patrickkwang — APPROVED** [review]

**Pull request merged**